### DISCUSSION TITLE
Add support for IP whitelisting

### DISCUSSION THREAD

**Discussion by ItsMe-Haris** [pocketbase/pocketbase#1403] [Q&A] [answered]
It would be a great addition and can help with API security. By default, it could be set to 0.0.0.0 while allows request from any IP and you can modify it from settings to only respond to specific IP/s.

**Please add this to the roadmap.**

In the meantime, is it possible to get IP address from `@request`?

**Comment by ganigeorgiev** [comment, marked as answer]
Similar to the https://github.com/pocketbase/pocketbase/issues/1232, you can do this if you [use PocketBase as framework](https://pocketbase.io/docs/use-as-framework/) OR use a reverse proxy like nginx, apache, caddy, etc.

> In the meantime, is it possible to get IP address from @request?

No, because fetching the "real" client IP is not always an easy task in case you are behind a cdn(s) and or reverse-proxy(ies) and it may require some upfront setup to make it work properly.

**Reply by ItsMe-Haris** [reply]
Alright!

Another question, is it possible to access custom query parameters beside the predefined ones?
For example:
```
/api/collections/posts/records?something=xyz
```
is it possible to access value of `something` in API rules section?

**Reply by ganigeorgiev** [reply]
> is it possible to access value of something in API rules section?

Yes, you should be able to access it via the `@request.query.something` filter. For more info you can check https://pocketbase.io/docs/api-rules-and-filters/.

**Reply by ItsMe-Haris** [reply]
Perfect! Thanks.
